discord/app/events.py

```diff
@@ -0,0 +1,98 @@
+"""
+The MIT License (MIT)
+
+Copyright (c) 2021-present Pycord Development
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+
+from asyncio import Future
+import asyncio
+from typing import Any, Callable, Protocol, Self, Type, TypeVar
+
+from .state import ConnectionState
+
+T = TypeVar('T')
+
+
+class Event(Protocol):
+    __event_name__: str
+
+    @classmethod
+    async def __load__(cls, data: dict[str, Any], state: ConnectionState) -> Self:
+        ...
+
+
+class EventEmitter:
+    def __init__(self, state: ConnectionState) -> None:
+        self._listeners: dict[Type[Event], list[Callable]] = {}
```

Comment:
`Type` is [deprecated](https://docs.python.org/3/library/typing.html#typing.Type), use `type` (builtin)